src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -628,6 +628,49 @@ public bool Remove(TKey key)
             return false;
         }
 
+        public bool Remove(TKey key, out TValue value)
+        {
+            if (key == null)
+            {
+                value = default(TValue);
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.key);
+            }
+
+            if (buckets != null)
```

Comment:
I am not sure whether it would be a win. You would need to collect throughput and size data to tell what is better. I am ok with the implementation in one big method as is.

I would rather look into how to make the code for this implementation smaller. For example, caching pointer to `entries[i]` in a local may help to reduce native code size speed and `ref Entry entry = ref entries[i];`